docs/developer-guide/plugins-scenarios.md

```diff
@@ -4,258 +4,342 @@
+        return chain.proceed(request)
+                .map(response -> response.newBuilder()
+                        .header("X-Foo", "bar")
+                        .build()
+                );
+    }
+}
+```
 	
 ### Asynchronously transform request object
 
-Sometimes it is necessary to transform the request asynchronously. For example, it may be necessary to consult an 
-external service(s) like databases and such to look up necessary information for the request transformation. In this case:
-1. Perform a non-blocking call to an asynchronous API. It is best to wrap such API behind Rx Java observables.
-2. Call the chain.proceed(request) when the the asynchronous operation completes.
-3. Because chain.proceed() itself is an asynchronous operation returning Observable[HttpResponse], it needs to be called from inside Rx flatMap() operator.
-
-In the following example, the interceptor either allows or rejects the request based on a query to an external service.
-The API for external service is wrapped behind requestAccessQueryAsynch() method. Normally a method like this would 
-initiate a transaction to the external service, and return a Future, Promise, Rx Observable, etc. of the outcome. 
-For sake of simplicity, here it returns an Observable.just(true).
-
-The flatMap() operator allows us to register a function that itself performs an asynchronous operation. In our example, 
-requestAccessQueryAsynch() returns back an Observable[Boolean] indicating if the access is granted or not. When the 
-access query completes, Rx framework calls out to the registered function, which in this example is a lambda expression 
-of type Boolean => Observable[HttpResponse]. The lambda expression transforms the access grant outcome into a HttpResponse 
-object. This is an asynchronous transformation, because the lambda expression calls out to asynchronous chain.proceed().
-
-Our lambda expression looks into the access grant outcome. When true, the request is proxied onwards by a call to 
-chain.proceed(request). Otherwise a FORBIDDEN response is returned.
-	
-	import com.hotels.styx.api.HttpInterceptor;
-	import com.hotels.styx.api.HttpRequest;
-	import com.hotels.styx.api.HttpResponse;
-	...
-	 
-    public class AsyncRequestDelayInterceptor implements HttpInterceptor {
-   	    private static final HttpResponse rejected = HttpResponse.Builder.response(FORBIDDEN).build();
-        @Override
-        public Observable<HttpResponse> intercept(HttpRequest request, Chain chain) {
-            return requestAccessQueryAsync(request)
-                .flatMap((allowed) -> allowed ? chain.proceed(request) : Observable.just(rejected));
-        }
+Sometimes it is necessary to transform the request asynchronously. For example, may need to 
+look up external key value stores to parametrise the request transformation. The example below
+shows how to modify a request URL path based on an asynchronous lookup to an external database.
```

Comment:
It could mean non-blocking activities. Or it could mean offloading some operations to remote threads. 

I hope our audience is familiar with the term. But if not we can always clarify it later. But for now I would like to get the bulk of the text in, plus all other important styx 1.0 work as a priority.